athena-gcs/src/test/java/com/amazonaws/athena/connectors/gcs/filter/FilterExpressionBuilderTest.java

```diff
@@ -0,0 +1,82 @@
+
+import com.amazonaws.athena.connector.lambda.data.Block;
+import com.amazonaws.athena.connector.lambda.data.BlockAllocatorImpl;
+import com.amazonaws.athena.connector.lambda.data.SchemaBuilder;
+import com.amazonaws.athena.connector.lambda.domain.predicate.Constraints;
+import com.amazonaws.athena.connector.lambda.domain.predicate.Marker;
+import com.amazonaws.athena.connector.lambda.domain.predicate.Range;
+import com.amazonaws.athena.connector.lambda.domain.predicate.SortedRangeSet;
+import com.amazonaws.athena.connector.lambda.domain.predicate.ValueSet;
+import com.amazonaws.athena.connector.lambda.security.FederatedIdentity;
+import com.amazonaws.athena.connectors.gcs.GcsTestUtils;
+import org.apache.arrow.vector.complex.reader.FieldReader;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.anyString;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore({"com.sun.org.apache.xerces.*", "javax.xml.*", "org.xml.*",
+        "javax.management.*", "org.w3c.*", "javax.net.ssl.*", "sun.security.*", "jdk.internal.reflect.*", "javax.crypto.*"
+})
+@PrepareForTest({GcsTestUtils.class})
+public class FilterExpressionBuilderTest
+{
+    private static final String STORAGE_SPLIT_JSON = "storage_split_json";
+    @Mock
+    public FederatedIdentity federatedIdentity;
+
+    @Test
+    public void testGetExpressions()
+    {
+        Schema schema = SchemaBuilder.newBuilder().addField("id", new ArrowType.Int(64, false)).build();
+        FilterExpressionBuilder filterExpressionBuilder = new FilterExpressionBuilder(schema);
+        List<AbstractExpression> exp = filterExpressionBuilder.getExpressions(new Constraints(createSummaryWithLValueRangeEqual("id", new ArrowType.Int(64, false), 1L)));
+        assertNotNull(exp);
+    }
+
+    public static Map<String, ValueSet> createSummaryWithLValueRangeEqual(String fieldName, ArrowType fieldType, Object fieldValue)
+    {
+        Block block = Mockito.mock(Block.class);
+        FieldReader fieldReader = Mockito.mock(FieldReader.class);
+        Mockito.when(fieldReader.getField()).thenReturn(Field.nullable(fieldName, fieldType));
+
+        Mockito.when(block.getFieldReader(anyString())).thenReturn(fieldReader);
+        Marker low = Marker.exactly(new BlockAllocatorImpl(), new ArrowType.Int(32, true), fieldValue);
+        return Map.of(
+                fieldName, SortedRangeSet.of(false, new Range(low, low))
+        );
```

Comment:
I could be wrong, but as far as I understand the code so far, these constraints are only being applied to string partition columns right?

If so, do the tests need to be updated to test with Strings? There are a bunch of locations across the tests that are testing `SortedRangeSet` with `ArrowType.Int(32, true)` but I'm not sure why?

Reply:
when I run below query : 
SELECT * FROM "partition_layout_test"."hive_nonhive_test_1" where year = '2000';
we get below constrains in read request:
![image](https://user-images.githubusercontent.com/108925344/214036124-b867f7e2-7cb9-4d09-a2e0-194c15181839.png)

I observed that every constarins is instance of SortedRangeSet.

Reply:
Thanks for the confirmation and screenshot.